packages/app-mobile/root.tsx

```diff
@@ -982,7 +982,8 @@ class AppComponent extends React.Component {
 		);
 		onSystemColorSchemeChange(Appearance.getColorScheme());
 
-		setupQuickActions(this.props.dispatch, this.props.selectedFolderId);
+		await setupQuickActions(this.props.dispatch);
+		this.deviceEventerEmitter_ = DeviceEventEmitter.addListener('quickActionShortcut', quickActionHandler(this.props.dispatch));
```

Comment:
Let's try to minimise that kind of code as much as possible. So please make setupQuickAction register the listener, and make it return the listener. Please name it `quickActionShortcutListener_`. Then use this when the component unmount to remove the listener.

So here you'll only have `this.quickActionShortcutListener_ = await setupQuickActions(this.props.dispatch);`

Reply:
My bad, I didn't even try to do that because I thought it wouldn't work. I thought the async calls used for `QuickActions` wouldn't allow me to return something at the end of the function. Not really sure what is going on, but it is working.